src/Core/Content/Cookie/SalesChannel/CookieRoute.php

```diff
@@ -35,7 +41,32 @@ public function getDecorated(): AbstractCookieRoute
     public function getCookieGroups(Request $request, SalesChannelContext $salesChannelContext): CookieRouteResponse
     {
         $cookieGroups = $this->cookieProvider->getCookieGroups($salesChannelContext);
+        $hash = $this->generateCookieConfigurationHash($cookieGroups);
```

Comment:
```suggestion
        $hash = $this->generateCookieConfigurationHash(clone $cookieGroups);
```

I think something like this is also possible. then it maybe more clearer if you read the method call first